ql/src/java/org/apache/hadoop/hive/ql/cache/results/QueryResultsCache.java

```diff
@@ -319,9 +312,8 @@ public boolean waitForValidStatus() {
             // Status has not changed, continue waiting.
             break;
           }
-
           synchronized (this) {
-            this.wait(timeout);
+            this.wait();
```

Comment:
This thread is waiting on the state change.  Adding a timeout doesn't cause the loop to exit, instead the thread will just loop and see that there was no change and `wait` again.  This timeout only adds churn.